torch/fx/experimental/symbolic_shapes.py

```diff
@@ -1684,30 +1686,36 @@ def create_symbol(
             # If we're not duck shaping, we always create a new symbol
             # Even if we're duck shaping, if we haven't seen this particular
             # value before, we also create a new symbol
-            sympy_expr = sympy.Symbol(f"s{len(self.var_to_val)}", positive=True, integer=True)
-            self.log.info("create_symbol %s = %s for %s", sympy_expr, val, source.name())
-            # We always associate vars to vals
-            self.var_to_val[sympy_expr] = sympy.Integer(val)
+            if isinstance(val, int):
+                sympy_expr = sympy.Symbol(f"s{len(self.var_to_val)}", positive=True, integer=True)
+                self.log.info("create_symbol %s = %s for %s", sympy_expr, val, source.name())
+                # We always associate vars to vals
+                self.var_to_val[sympy_expr] = sympy.Integer(val)
+                # Apply default range, which assumes not zero-one
+                self.var_to_range[sympy_expr] = self._default_value_range()
+                # Small performance optimization: if we have a min-max constraint,
+                # we can proactively narrow to that range
+                if isinstance(constraint_dim, StrictMinMaxConstraint):
+                    assert not duck
+                    self.var_to_range[sympy_expr] &= constraint_dim.vr
+
+                vr = self.var_to_range[sympy_expr]
+                if val not in vr:
+                    raise RuntimeError(f"{val} not in range [{vr.lower}, {vr.upper}]")
+            elif isinstance(val, tuple):
```

Comment:
Hacking in support for tuple-backed SymInts using `sympy.Array`?? (tons more changes needed to do this right)

NB: tuple here because lists aren't hashable but tuples are